src/flows/Onboarding/hooks.ts

```diff
@@ -143,6 +150,55 @@ const useJSONSchemaForm = ({
   });
 };
 
+const useBenefitOffersSchema = (
+  employmentId: string,
+  options: OnboardingHookProps['options'],
+) => {
+  const jsonSchemaQueryParam = options?.jsonSchemaVersion
+    ?.benefit_offers_form_schema
+    ? {
+        json_schema_version:
+          options.jsonSchemaVersion.benefit_offers_form_schema,
+      }
+    : {};
+  const { client } = useClient();
+  return useQuery({
+    queryKey: ['benefit-offers-schema', employmentId],
+    retry: false,
+    enabled: !!employmentId,
+    queryFn: async () => {
+      const response = await getShowSchema({
+        client: client as Client,
+        headers: {
+          Authorization: ``,
+        },
+        path: {
+          employment_id: employmentId,
+        },
+        query: jsonSchemaQueryParam,
+      });
+
+      // If response status is 404 or other error, throw an error to trigger isError
+      if (response.error || !response.data) {
+        throw new Error('Failed to fetch benefit offers schema');
+      }
+
+      return response;
+    },
+    select: ({ data }) => {
+      const { schema } = modify(
+        data.data?.schema || {},
+        options?.jsfModify || {},
+      );
```

Comment:
It's probably ok to do this but I would rather execute `modify` only `options?.jsfModify` is defined.